[explicit-member-accessibility] False positive when using private fields

**Repro**

The rule `@typescript-eslint/explicit-member-accessibility` reports a false positive when using private fields. At the same time, TypeScript reports the error `An accessibility modifier cannot be used with a private identifier` if an accessibility modifier would be used.

```JSON
{
  "rules": {
    "@typescript-eslint/explicit-member-accessibility": ["warn"]
  }
}
```

```TS
class Person {
    #name: string

    constructor(name: string) {
        this.#name = name;
    }

    greet() {
        console.log(`Hello, my name is ${this.#name}!`);
    }
}
```

**Expected Result**

No false positive when using private fields.

**Actual Result**

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.22.0` |
| `@typescript-eslint/parser`        | `2.22.0` |
| `TypeScript`                       | `3.8.3` |
| `ESLint`                           | `6.8.0` |
| `node`                             | `12.16.1` |
| `npm`                              | `6.14.1` |

I've just updated to `eslint@^8.2.0` and I'm still getting this error

|package|version|
|-|-|
|eslint|8.2.0|
|@typescript-eslint/parser|5.3.0|
|@typescript-eslint/eslint-plugin|5.3.0|
|typescript|4.3.5|
|node|16.4.0|

```
   26:2  error  Missing accessibility modifier on class property #length                     @typescript-eslint/explicit-member-accessibility
```

Do I need to do something to make this work?

--- a/packages/eslint-plugin/src/rules/explicit-member-accessibility.ts
+++ b/packages/eslint-plugin/src/rules/explicit-member-accessibility.ts
@@ -214,6 +214,10 @@ export default util.createRule<Options, MessageIds>({
         | TSESTree.PropertyDefinition
         | TSESTree.TSAbstractPropertyDefinition,
     ): void {
+      if (propertyDefinition.key.type === AST_NODE_TYPES.PrivateIdentifier) {
+        return;
+      }
+
       const nodeType = 'class property';
 
       const { name: propertyName } = util.getNameFromMember(

--- a/packages/eslint-plugin/tests/rules/explicit-member-accessibility.test.ts
+++ b/packages/eslint-plugin/tests/rules/explicit-member-accessibility.test.ts
@@ -342,7 +342,7 @@ class Test {
   #bar() {}
 }
       `,
-      options: [{ accessibility: 'no-public' }],
+      options: [{ accessibility: 'explicit' }],
     },
   ],
   invalid: [

